app.js

```diff
@@ -1,38 +1,39 @@
 
 // Get Flat Number from API
-function validateCall() {
-  fetch('https://next.json-generator.com/api/json/get/41_QW6ze5')
-  .then(res => res.json())
-  .then(res => {
-    let flatNumbers = [];
-    let dial = screenPhone.textContent;
-    let numDial = parseInt(dial);
-    for (let i = 0; i < res.length; i++) {
-      flatNumbers.push(res[i].number)
-    };
-    let indxNumber = flatNumbers.indexOf(numDial)
-    
+const url = 'https://next.json-generator.com/api/json/get/41_QW6ze5';
+
+// new vertion
 
-    if (screenPhone.textContent == '') {
-      AlertMessage('Ingrese el numero de un apartamento', 'error');
+function validateNumber() {
+  fetch(url)
+  .then(res => res.json())
+  .then(result => {
+    const apartmentNumber = parseInt(screen.textContent);
+    if(screen.textContent === '') {
+      AlertMessage('Ingrese un numero antes de llamar', 'error');
     } else {
-      if (flatNumbers.includes(numDial)){
+      apartmentData = result.find((apartment) => apartment.number === apartmentNumber);
+      if(apartmentData !== undefined){
+        validateStatus(apartmentData)
+        console.log(apartmentData)
         AlertMessage('Marcando', 'dialing');
-        let flatOwner = res[indxNumber].owner
-        
-        if (res[indxNumber].status == 'ANSWERED')
-          setTimeout(()=>{AlertMessage(`${flatOwner.firstName} ${flatOwner.lastName} ha permitido su entrada `, 'succes')}, 3000);
-          
-        else {
-          setTimeout(()=>{AlertMessage('Sin respuesta, no puede ingresar', 'error')}, 3000);
-        }
+
       } else {
-        AlertMessage('No es posible llamar, el apartamento no existe', 'error')
-      };
+        AlertMessage('No es posible llamar, el apartamento no existe', 'error');
+      }
     }
-  }) 
+  })  
 };
 
+function validateStatus(data) {
```

Comment:
what if data is undefined?
read about **Guard Clause**
https://medium.com/better-programming/refactoring-guard-clauses-2ceeaa1a9da